source/common/upstream/health_checker_impl.cc

```diff
@@ -84,6 +107,23 @@ std::chrono::milliseconds HealthCheckerImplBase::interval() {
   return std::chrono::milliseconds(final_ms);
 }
 
+void HealthCheckerImplBase::addHosts(const std::vector<HostSharedPtr>& hosts) {
+  for (const HostSharedPtr& host : hosts) {
+    active_sessions_[host] = makeSession(host);
+    active_sessions_[host]->start();
+  }
+}
+
+void HealthCheckerImplBase::onClusterMemberUpdate(const std::vector<HostSharedPtr>& hosts_added,
+                                                  const std::vector<HostSharedPtr>& hosts_removed) {
+  addHosts(hosts_added);
+  for (const HostSharedPtr& host : hosts_removed) {
+    auto session_iter = active_sessions_.find(host);
+    ASSERT(active_sessions_.end() != session_iter);
+    active_sessions_.erase(session_iter);
```

Comment:
If this ASSERT is false then the erase() has undefined behavior.  Should it be a RELEASE_ASSERT or otherwise inhibit call to erase() if not found in opt builds?